### ISSUE TITLE
samtools view cannot read gzip sam (1.9->1.10 change)

### ISSUE THREAD

**Issue by keiranmraine** [samtools/samtools#1286]
I don't think I see anything in the release notes that indicates this behaviour was removed, but perhaps it wasn't officially supported.

Create a gzip compressed sam file:

```bash
samtools view -H data.bam > tmp.sam
samtools view data.bam | head >> tmp.sam
gzip tmp
```

read with samtools 1.9:

```bash
$ module load samtools/1.9
$ samtools view -H tmp.gz | wc -l 
3376
```

Fail to reads with samtools 1.10

```bash
$ samtools view -H tmp.gz | wc -l 
0
$ samtools view -H tmp.gz
Segmentation fault (core dumped)
# forcing sam doesn't change this
$ samtools view -SH tmp.gz
Segmentation fault (core dumped)
$ zcat tmp.gz | samtools view -H - | wc -l 
3377
```

**Comment by jmarshall** [comment]
Try it also with **develop**; I expect this was fixed by samtools/htslib#1019.

**Comment by keiranmraine** [comment]
I can confirm the develop branch of samtools (with develop of htslib) handles this correctly.
